lib/experimental/fonts/font-library/class-wp-rest-font-families-controller.php

```diff
@@ -0,0 +1,503 @@
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+
+		$font_family = new WP_Font_Family( array( 'slug' => $request['slug'] ) );
+		$result      = $font_family->uninstall();
+
+		if ( is_wp_error( $result ) ) {
+			if ( 'font_family_slug_not_found' === $result->get_error_code() ) {
+				$result->add_data( array( 'status' => 404 ) );
+			} else {
+				$result->add_data( array( 'status' => 500 ) );
+			}
+		}
+		return rest_ensure_response( $result );
+	}
+
+	/**
+	 * Checks whether the user has permissions to update the Font Library.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
+	 */
+	public function update_font_library_permissions_check() {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_update_font_library',
+				__( 'Sorry, you are not allowed to update the Font Library on this site.' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Checks whether the user has write permissions to the temp and fonts directories.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return true|WP_Error True if the user has write permissions, WP_Error object otherwise.
+	 */
+	private function has_write_permission() {
+		// The update endpoints requires write access to the temp and the fonts directories.
+		$temp_dir   = get_temp_dir();
+		$upload_dir = WP_Font_Library::get_fonts_dir();
+		if ( ! is_writable( $temp_dir ) || ! wp_is_writable( $upload_dir ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Checks whether the request needs write permissions.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $font_family Font family definition.
+	 * @return bool Whether the request needs write permissions.
+	 */
+	private function needs_write_permission( $font_family ) {
+		if ( isset( $font_family['fontFace'] ) ) {
+			foreach ( $font_family['fontFace'] as $face ) {
+				// If the font is being downloaded from a URL or uploaded, it needs write permissions.
+				if ( isset( $face['downloadFromUrl'] ) || isset( $face['uploadedFile'] ) ) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Installs new fonts.
+	 *
+	 * Takes a request containing new fonts to install, downloads their assets, and adds them
+	 * to the Font Library.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request The request object containing the new fonts to install
+	 *                                 in the request parameters.
+	 * @return WP_REST_Response|WP_Error The updated Font Library post content.
+	 */
+	public function create_item( $request ) {
+		$font_family_data = array(
+			'slug'       => $request->get_param( 'slug' ),
+			'name'       => $request->get_param( 'name' ),
+			'fontFamily' => $request->get_param( 'fontFamily' ),
```

Comment:
All request arguments should be snake case, not camelcase

Reply:
Updated.  
Note that data being passed around is still camelCase (so as to align with theme.json style data structure) however the API arguments are now all snake case.